Regular update 2021-05-04

romainx opened the pull request:
Hello,

- Few packages updated
- Fix Hadolint warnings
- Style

Not updated Mambaforge version since it brings a new version of Python `3.9` thanks @mathbunnyru for the heads-up.

- Added a test to check the Python version to avoid this kind of surprise in the future.

Best.

mathbunnyru commented on base-notebook/Dockerfile:
```diff
@@ -51,10 +51,9 @@ RUN apt-get -q update \
- && apt-get clean && rm -rf /var/lib/apt/lists/*
-
-RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
+    run-one && \
```
Please, fix the style for `&&` several lines above as well.

romainx replied:
Hello,

I do not understand what need to be fixed here regarding the style, sorry.

```dockerfile
fonts-liberation \
run-one && \
apt-get clean && rm -rf /var/lib/apt/lists/* && \
echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
locale-gen
```

mathbunnyru replied:
Several lines above the line start from `&&`like this: `&& apt-get install -yq --no-install-recommends \`

romainx replied:
Ok got it, I though the comment was related to my change, but Ok I've seen it.

mathbunnyru replied:
Thanks :)
Sorry for being not 100% clear.

mathbunnyru commented on base-notebook/Dockerfile:
```diff
@@ -51,10 +51,9 @@ RUN apt-get -q update \
-
-RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
+    run-one && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* && \
+    echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
```
I think having separate steps is actually good in this case (there is a clear distinction in the processes), so I would prefer hadolint ignore for this line.

mathbunnyru replied:
After a closer look, it actually makes sense both ways, because several lines above we install locales. So, I'm ok with this change.

romainx replied:
So if you are Ok :-)

mathbunnyru commented on base-notebook/Dockerfile:
```diff
@@ -24,7 +24,7 @@ USER root
 # Check https://github.com/conda-forge/miniforge/releases
 # Conda version
-ARG conda_version="4.10.0"
+ARG conda_version="4.10.1"
```
This change is not actually that simple.
MiniForge has started using python 3.9, so with this change we will switch to python 3.9 as well.
You can see it in the tagging step :)

There is a small release note as well.
https://github.com/conda-forge/miniforge/releases

mathbunnyru replied:
Maybe we should wait at least till anaconda ships python 3.9 by default?

romainx replied:
I did not even imagine that the version of Python could change in a patch release. If you want to master the python version and avoid an unwanted change in the future I think we have to write a test. I can do it.

romainx replied:
For information, [Miniconda](https://docs.conda.io/en/latest/miniconda.html) (the previous installer is shipped also with Python 3.9 `Miniconda3-py39_4.9.2-Linux-x86_64.sh`.
What was applied here until now was to always use the latest versions of the software and to rely on tags if users want to use a previous version.

mathbunnyru replied:
I was also surprised by changing of the Python version in a patch release.

Yes, let's write a test, which detects the current python version and compares it with hardcoded constant - this way we will always know when python is updated.

Ok, let's write a test and update.

romainx replied:
Sure I will do that 👍

romainx replied:
Test implemented but I rolled back the version of Python to stick to `3.8.x`. We will update it later.

mathbunnyru reviewed (changes requested):
Let's think about updating python.

romainx commented on tensorflow-notebook/Dockerfile:
```diff
@@ -7,8 +7,7 @@ LABEL maintainer="Jupyter Project <jupyter@googlegroups.com>"
 # Install Tensorflow
 RUN mamba install --quiet --yes \
-    'tensorflow=2.4.1' \
-    && \
```
@mathbunnyru is it the style you want to use? If it's the case I have fixed it.

mathbunnyru replied:
No, I added the comment about the line I didn't like.

romainx replied:
Ok because I've rolled back your previous style change and I made the link with your other remark on style with `&&` at the beginning of a line. So I would like to be sure.

mathbunnyru replied:
Thanks, right now the style looks good and consistent :)

romainx commented:
I'm converted it do draft while waiting for @mathbunnyru approval.

romainx commented:
@mathbunnyru I've implemented all the changes and also reverted to keep Python `3.8.x`. Please tell me if it's Ok for you now and I will merge it. Thanks.

mathbunnyru commented on base-notebook/test/test_python.py:
```diff
@@ -0,0 +1,19 @@
+
+
+def test_python_version(container, python_max_version="3.9"):
+    """Check that python version is not higher than a max version"""
```
```suggestion
    """Check that python version is lower than a max version"""
```

I would also change the word max to the word next.

romainx replied:
Docstring changed, also change the word max to next as you suggested. I hope it will be Ok now.

mathbunnyru reviewed (approved):
👍

romainx commented:
Ok thanks, I will merge it. 
@mathbunnyru I'm gonna take a week break or a week and a half break, I will be able to spend a limited time here before so if something needs to be reviewed and merged before, tell me.

mathbunnyru commented:
> Ok thanks, I will merge it.
> @mathbunnyru I'm gonna take a week break or a week and a half break, I will be able to spend a limited time here before so if something needs to be reviewed and merged, tell me.

Could you please take a look in the PR: https://github.com/jupyter/docker-stacks/pull/1277?

mathbunnyru commented:
Another PR, which is ready (you already took a look and it should be good to go): https://github.com/jupyter/docker-stacks/pull/1288

So, two reviews in total, which can be easily merged.
